File: app/src/main/java/dev/ragnarok/fenrir/db/interfaces/IVideoAlbumsStorage.java
package dev.ragnarok.fenrir.db.interfaces;

import androidx.annotation.NonNull;

import java.util.List;

import dev.ragnarok.fenrir.db.model.entity.VideoAlbumEntity;
import dev.ragnarok.fenrir.model.VideoAlbumCriteria;
import io.reactivex.rxjava3.core.Completable;
import io.reactivex.rxjava3.core.Single;


public interface IVideoAlbumsStorage extends IStorage {
    Single<List<VideoAlbumEntity>> findByCriteria(@NonNull VideoAlbumCriteria criteria);

    Completable insertData(int accountId, int ownerId, @NonNull List<VideoAlbumEntity> data, boolean invalidateBefore);
}

File: app/src/main/kotlin/dev/ragnarok/fenrir/db/interfaces/IVideoAlbumsStorage.kt
package dev.ragnarok.fenrir.db.interfaces

import dev.ragnarok.fenrir.db.model.entity.VideoAlbumDboEntity
import dev.ragnarok.fenrir.model.VideoAlbumCriteria
import io.reactivex.rxjava3.core.Completable
import io.reactivex.rxjava3.core.Single

interface IVideoAlbumsStorage : IStorage {
    fun findByCriteria(criteria: VideoAlbumCriteria): Single<List<VideoAlbumDboEntity>>
    fun insertData(
        accountId: Int,
        ownerId: Int,
        data: List<VideoAlbumDboEntity>,
        invalidateBefore: Boolean
    ): Completable
}